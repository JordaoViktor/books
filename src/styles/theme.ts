export default {
  colors: {
    primary: "#FFFF",
    secondary: "#333333",
    tertiary: "#AB2680",
    background: "#E5E5E5",
    dark: "#000",

    lightenText: "#FFFF",
    darkText: "#333333",

    darkOpacity200: "rgba(0,0,0,0.20)",
    darkOpacity300: "rgba(0,0,0,0.32)",
    darkOpacity400: "rgba(0,0,0,0.4)",

    transparent: "transparent"
  },
  fonts: {
    light: "Heebo_300Light",
    normal: "Heebo_400Regular",
    bold: "Heebo_500Medium",
  },
}