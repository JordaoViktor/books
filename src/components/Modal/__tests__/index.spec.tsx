import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import theme from '@styles/theme'
import { Modal } from '../index';

interface IProviderProps {
  children?: React.ReactNode,
}

const Provider = ({ children }: IProviderProps) => (
  <ThemeProvider theme={theme} >
    {children}
  </ThemeProvider>
);

describe('Input Component', () => {
  test('matches modal snapshot', () => {
    const onRequestCloseMock = jest.fn();

    const modalRender = renderer
      .create(
        <ThemeProvider theme={theme} >
          <Modal visible={true} onRequestClose={onRequestCloseMock} />
        </ThemeProvider>)
      .toJSON();

    expect(modalRender).toMatchSnapshot();

  });

  it('ensure modal receives visible prop false', () => {
    const onRequestCloseMock = jest.fn();
    const { getByTestId } = render(<Modal visible={false} onRequestClose={onRequestCloseMock} />, { wrapper: Provider })

    const modalComponent = getByTestId('modal')

    expect(modalComponent.props.visible).toBeFalsy()
  })

  it('ensure button receives do not some color', () => {
    const onRequestCloseMock = jest.fn();
    const { getByTestId } = render(<Modal visible={true} onRequestClose={onRequestCloseMock} />, { wrapper: Provider })

    const modalComponent = getByTestId('modal')

    expect(modalComponent.props.visible).toBeTruthy()
  })
})