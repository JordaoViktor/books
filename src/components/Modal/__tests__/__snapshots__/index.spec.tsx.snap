// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component matches modal snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[MockFunction]}
  style={
    Array [
      Object {},
    ]
  }
  testID="modal"
  transparent={true}
  visible={true}
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.4)",
          "height": "100%",
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#FFFF",
            "borderRadius": 4,
            "marginTop": 50,
            "minHeight": "70%",
            "width": "90%",
          },
        ]
      }
    />
  </RCTSafeAreaView>
</Modal>
`;
